internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,240 @@
+}
+
+type TransactionServiceOptions struct {
+	DistributionAccountSignatureClient signing.SignatureClient
+	ChannelAccountSignatureClient      signing.SignatureClient
+	HorizonClient                      horizonclient.ClientInterface
+	RpcUrl                             string
+	BaseFee                            int64
+	Ctx                                context.Context
+}
+
+func (o *TransactionServiceOptions) ValidateOptions() error {
+	if o.DistributionAccountSignatureClient == nil {
+		return fmt.Errorf("distribution account signature client cannot be nil")
+	}
+
+	if o.ChannelAccountSignatureClient == nil {
+		return fmt.Errorf("channel account signature client cannot be nil")
+	}
+
+	if o.HorizonClient == nil {
+		return fmt.Errorf("horizon client cannot be nil")
+	}
+
+	if o.RpcUrl == "" {
+		return fmt.Errorf("rpc url cannot be empty")
+	}
+
+	if o.BaseFee < int64(txnbuild.MinBaseFee) {
+		return fmt.Errorf("base fee is lower than the minimum network fee")
+	}
+	return nil
+}
+
+func NewTransactionService(opts TransactionServiceOptions) (TransactionService, error) {
+	if err := opts.ValidateOptions(); err != nil {
+		return nil, err
+	}
+	return &transactionService{
+		DistributionAccountSignatureClient: opts.DistributionAccountSignatureClient,
+		ChannelAccountSignatureClient:      opts.ChannelAccountSignatureClient,
+		HorizonClient:                      opts.HorizonClient,
+		RpcUrl:                             opts.RpcUrl,
+		BaseFee:                            opts.BaseFee,
+		Ctx:                                opts.Ctx,
+	}, nil
+}
+
+func parseJSONBody(body []byte) (map[string]interface{}, error) {
+	var res map[string]interface{}
+	err := json.Unmarshal(body, &res)
+	if err != nil {
+		return nil, fmt.Errorf(err.Error())
+	}
+	return res, nil
+}
+
+func parseErrorResultXdr(errorResultXdr string) (tss.RPCTXCode, error) {
+	errorResult := xdr.TransactionResult{}
+	err := errorResult.UnmarshalBinary([]byte(errorResultXdr))
+
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf("SendTransaction: unable to unmarshal errorResultXdr: %s", errorResultXdr)
+	}
+	return tss.RPCTXCode{
+		TxResultCode: errorResult.Result.Code,
+	}, nil
+}
+
+func sendRPCRequest(rpcUrl string, method string, params map[string]string) (map[string]interface{}, error) {
+	payload := map[string]interface{}{
+		"jsonrpc": "2.0",
+		"id":      1,
+		"method":  method,
+		"params":  params,
+	}
+	jsonData, _ := json.Marshal(payload)
+
+	resp, err := RpcPost(rpcUrl, "application/json", bytes.NewBuffer(jsonData))
+	if err != nil {
+		return nil, fmt.Errorf(method+": sending POST request to rpc: %v", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := UnMarshalRPCResponse(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf(method+": unmarshalling rpc response: %v", err)
+	}
+	res, err := UnMarshalJSON(body)
+	if err != nil {
+		return nil, fmt.Errorf(method+": parsing rpc response JSON: %v", err)
```

Comment:
Let's be consistent
```suggestion
		return nil, fmt.Errorf(method+": parsing rpc response json: %v", err)
```
or 
```suggestion
		return nil, fmt.Errorf(method+": parsing RPC response JSON: %v", err)
```

Reply:
Done